ale_linters/kotlin/ktlint.vim

```diff
@@ -0,0 +1,71 @@
+" Author: Francis Agyapong <francisagyapong2@gmail.com>
+" Description: Lint kotlin files using ktlint
+
+let g:ale_kotlin_ktlint_exec = get(g:, 'ale_kotlin_ktlint_exec', 'ktlint')
+let g:ale_kotlin_ktlint_rulesets = get(g:, 'ale_kotlin_ktlint_rulesets', [])
+let g:ale_kotlin_ktlint_format = get(g: , 'ale_kotlin_ktlint_format', 0)
+let g:ale_kotlin_ktlint_config_file = get(g:, 'ale_kotlin_ktlint_config_file', '.ale_kotlin_ktlint_config')
+let g:ale_kotlin_ktlint_enable_config = get(g:, 'ale_kotlin_ktlint_enable_config', 0)
+
+
+function! ale_linters#kotlin#ktlint#GetCommand(buffer) abort
+    if ale#Var(a:buffer, 'kotlin_ktlint_enable_config')
+        let l:config_file = ale#path#FindNearestFile(a:buffer, ale#Var(a:buffer, 'kotlin_ktlint_config_file'))
```

Comment:
After reading the documentation for `g:ale_patterns_options`, I'm wondering how one will configure options for a specific directory. That is, how will options intended for a specific project (or directory) be set without having to add the directory to `g:ale_pattern_options`?